bpfman/src/lib.rs

```diff
@@ -1801,25 +1801,29 @@ fn detach_multi_attach_program(
     } else {
         false
     };
-    debug!("netns_deleted = {netns_deleted}");
+    let number_attached_programs = num_attached_programs(&did, root_db)?;
+    debug!(
+        "netns_deleted = {netns_deleted}, number_attached_programs = {number_attached_programs}"
+    );
 
-    let next_available_id = if netns_deleted {
+    // Typically, we'd expect the number of attached programs to be greater than
+    // 0 because we're calling the detach function.  However, if
+    // num_attached_programs() can't find the dispatcher for the given did, it
+    // will return 0, so we need to handle that case.
+    let next_available_id = if number_attached_programs == 0 {
```

Comment:
This isn't easy to read  even with the comment.
You could write this as:
```rust
let next_available_id = num_attached_programs(&did, root_db)?.saturating_sub(1);
```
That would require num_attached_programs to return a `u32` - which I don't think is unreasonable since we'd never expect a negative number of programs. IIRC it was a usize because we use it to index something at some point an wanted to avoid the `as` cast.

Reply:
Done.